app/models/course/material/folder.rb

```diff
@@ -19,7 +19,7 @@ def files_attributes=(files)
   end
 
   def self.after_course_initialize(course)
-    return if course.persisted?
+    return if course.persisted? || course.root_folder
```

Comment:
I think we need to define a `has_root_folder?`, because `course.root_folder` will give exception when not found